OpenDreamShared/Json/DreamCompiledJson.cs

```diff
@@ -1,15 +1,23 @@
 ﻿using System.Collections.Generic;
 
-namespace OpenDreamShared.Json {
-    public sealed class DreamCompiledJson {
-        public List<string>? Strings { get; set; }
-        public string[]? Resources { get; set; }
-        public int[]? GlobalProcs { get; set; }
-        public GlobalListJson? Globals { get; set; }
-        public ProcDefinitionJson? GlobalInitProc { get; set; }
-        public List<DreamMapJson>? Maps { get; set; }
-        public string? Interface { get; set; }
-        public DreamTypeJson[]? Types { get; set; }
-        public ProcDefinitionJson[]? Procs { get; set; }
-    }
+namespace OpenDreamShared.Json;
+
+public sealed class DreamCompiledJson {
+    public DreamCompiledJsonMetadata Metadata { get; set; }
+    public List<string>? Strings { get; set; }
+    public string[]? Resources { get; set; }
+    public int[]? GlobalProcs { get; set; }
+    public GlobalListJson? Globals { get; set; }
+    public ProcDefinitionJson? GlobalInitProc { get; set; }
+    public List<DreamMapJson>? Maps { get; set; }
+    public string? Interface { get; set; }
+    public DreamTypeJson[]? Types { get; set; }
+    public ProcDefinitionJson[]? Procs { get; set; }
+}
+
+public sealed class DreamCompiledJsonMetadata {
+    /// <summary>
+    ///  Hash of all the <c>DreamProcOpcode</c>s
+    /// </summary>
+    public string Version { get; set; }
```

Comment:
Could also change this to something like `OpcodeHash` - I figured Version as the field name was more clear.

Reply:
`Version` works well